src/model/entities/rundown.ts

```diff
@@ -559,22 +559,23 @@ export class Rundown extends BasicRundown {
    * If the Segment is currently OnAir, the Segment is still removed, but an "unsynced copy" is created of the Segment and added to the Rundown in its place.
    */
   public removeSegment(segmentId: string): Segment | undefined {
-    const segmentToRemove: Segment | undefined = this.segments.find(segment => !segment.isUnsynced() && segment.id === segmentId)
+    let segmentToRemove: Segment | undefined = this.segments.find(segment => !segment.isUnsynced() && segment.id === segmentId)
     if (!segmentToRemove) {
       return
     }
 
+    // TODO: Is it okay that i moved filtering up here? Otherwise we crash since we are looking for the unsynced segment id that has yet to exist.
```

Comment:
@LindvedKrvang I could use your opinion on this, i don't know if this could cause any issues elsewhere?

Reply:
It should be fine, but you could use the parameter `segmentId` as the filter instead of the the returned Segment from `this.unsyncSegment()`.

You could also do the thing we talked about.